Compile/interface/Filters/ZJetCutsFilter.h

```diff
@@ -326,9 +405,41 @@ class LeadingJetPtCut : public ZJetFilterBase
                        ZJetProduct const& product,
                        ZJetSettings const& settings) const override
     {
-        return (product.GetValidJetCount(settings, event) > 0)
-                ? (product.GetValidPrimaryJet(settings, event)->p4.Pt() > leadingJetPtMin)
-                : false;
+        LOG(DEBUG) << "\n[" << this->GetFilterId() << "]";
+        LOG(DEBUG) << "LeadingJetPtMin: " << leadingJetPtMin;
+        LOG(DEBUG) << "LEVEL: " << settings.GetCorrectionLevel();
+        int valids = product.m_validJets.size();
+        int jetcount = product.GetValidJetCount(settings, event, settings.GetCorrectionLevel());
+        LOG(DEBUG) << "Number of m_validJets= " << valids << ", Number of corrected jets: " << jetcount <<"\n";
+        if (valids != jetcount) {
+            LOG(ERROR) << "++++ Jet lost! Please investigate!! ++++";  // TODO!
```

Comment:
This can happen if we apply the JetEtaPhiCleaner, so why raise an error here? Wasn't this the issue we discussed last week? This should be fixes properly.

Reply:
Yes, good point!
Forgot to remove it.